platforms/x86-64/core/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![no_std]
+extern crate alloc;
+
+use core::time::Duration;
+use hal_core::{boot::BootInfo, PAddr};
+use hal_x86_64::cpu::local::GsLocalData;
+pub use hal_x86_64::{
+    cpu::{local::LocalKey, wait_for_interrupt},
+    mm,
+};
+use kernel::{mnemos_alloc::containers::Box, Kernel, KernelSettings};
+
+pub mod acpi;
+pub mod interrupt;
+
+pub fn init(bootinfo: &impl BootInfo, rsdp_addr: Option<PAddr>) -> &'static Kernel {
+    // TODO: init early tracing?
+    interrupt::enable_exceptions();
+    bootinfo.init_paging();
+
+    // TODO: init allocator!
```

Comment:
the bootinfo will give us a memory map we can use to populate a kernel allocator. this should provide something for `mnemos_alloc` but will also need to implement the [`page::Alloc` trait from mycelium hal](https://mycelium.elizas.website/hal_x86_64/mm/page/trait.alloc).

see https://github.com/hawkw/mycelium/blob/1f125194902cd4970b72eab0aa1d85d1b6ec1489/src/lib.rs#L118-L152

Reply:
if we want to init tracing before we init the allocator, we will probably need a small bump region for the `tracing` subscriber's single arc allocation.